pkg/arn/arnlike_test.go

```diff
@@ -0,0 +1,154 @@
+package arn
+
+import (
+	"strings"
+	"testing"
+)
+
+type arnLikeInput struct {
+	arn, pattern string
+}
+
+type quoteMetaInput struct {
+	input, expected string
+}
+
+func TestArnLikePostiveMatches(t *testing.T) {
+	inputs := []arnLikeInput{
+		{
+			arn:     `arn:aws:iam::000000000000:role/some-role`,
+			pattern: `arn:aws:iam::000000000000:role/some-role`,
+		},
+		{
+			arn:     `arn:aws:iam::000000000000:role/some-role`,
+			pattern: `arn:aws:iam::000000000000:*`,
+		},
+		{
+			arn:     `arn:aws:iam::000000000000:role/some-role`,
+			pattern: `arn:*:*:*:*:*`,
+		},
+		{
+			arn:     `arn:aws:iam::000000000000:role/some-role`,
+			pattern: `arn:aws:iam::000000000000:**`,
+		},
+		{
+			arn:     `arn:aws:iam::000000000000:role/some-role`,
+			pattern: `arn:aws:iam::000000000000:*role*`,
+		},
+		{
+			arn:     `arn:aws:iam::000000000000:role/some-role`,
+			pattern: `arn:aws:iam::000000000000:ro*`,
+		},
+		{
+			arn:     `arn:aws:iam::000000000000:role/some-role`,
+			pattern: `arn:aws:iam::000000000000:??????????????`,
+		},
+		{
+			arn:     `arn:aws:testservice::000000000000:some/wacky-new-[resource]{with}\metacharacters`,
+			pattern: `arn:aws:testservice::000000000000:some/wacky-new-[resource]{with}\metacharacters`,
+		},
+		{
+			arn:     `arn:aws:testservice::000000000000:some/wacky-new-[resource]{with}\metacharacters`,
+			pattern: `arn:aws:testservice::000000000000:some/wacky-new-[reso*`,
+		},
+	}
+
+	for _, v := range inputs {
+		ok, err := ArnLike(v.arn, v.pattern)
+		if err != nil {
+			t.Errorf("Expected no error for input arn: %s pattern: %s", v.arn, v.pattern)
+		}
+
+		if !ok {
+			t.Errorf("Expected true for input arn: %s pattern: %s", v.arn, v.pattern)
+		}
+	}
+}
+
+func TestArnLikeNetagiveMatches(t *testing.T) {
+	inputs := []arnLikeInput{
+		{
+			arn:     `arn:aws:iam::111111111111:role/some-role`,
+			pattern: `arn:aws:iam::000000000000:role/some-role`,
+		},
+		{
+			arn:     `arn:aws:testservice::000000000000:some/wacky:resource:with:colon:delims`,
+			pattern: `arn:aws:testservice::**:delims`,
+		},
+	}
+
+	for _, v := range inputs {
+		ok, err := ArnLike(v.arn, v.pattern)
+		if err != nil {
+			t.Errorf("Expected no error for input arn: %s pattern: %s", v.arn, v.pattern)
+		}
+
+		if ok {
+			t.Errorf("Expected false for input arn: %s pattern: %s", v.arn, v.pattern)
+		}
+	}
+}
+
+func TestArnLikeInvalidArns(t *testing.T) {
```

Comment:
Also test incomplete arnLikes: `arn:*` to ensure we are not allowing a glob to match the entire arn.

Reply:
Good call